stake-pool/js/src/instructions.ts

```diff
@@ -348,28 +448,42 @@ export class StakePoolInstruction {
       validatorVote,
       lamports,
       transientStakeSeed,
+      ephemeralStake,
+      ephemeralStakeSeed,
     } = params;
 
-    const type = STAKE_POOL_INSTRUCTION_LAYOUTS.IncreaseValidatorStake;
-    const data = encodeData(type, { lamports, transientStakeSeed });
-
     const keys = [
       { pubkey: stakePool, isSigner: false, isWritable: false },
       { pubkey: staker, isSigner: true, isWritable: false },
       { pubkey: withdrawAuthority, isSigner: false, isWritable: false },
       { pubkey: validatorList, isSigner: false, isWritable: true },
       { pubkey: reserveStake, isSigner: false, isWritable: true },
-      { pubkey: transientStake, isSigner: false, isWritable: true },
-      { pubkey: validatorStake, isSigner: false, isWritable: false },
-      { pubkey: validatorVote, isSigner: false, isWritable: false },
-      { pubkey: SYSVAR_CLOCK_PUBKEY, isSigner: false, isWritable: false },
-      { pubkey: SYSVAR_RENT_PUBKEY, isSigner: false, isWritable: false },
-      { pubkey: SYSVAR_STAKE_HISTORY_PUBKEY, isSigner: false, isWritable: false },
-      { pubkey: STAKE_CONFIG_ID, isSigner: false, isWritable: false },
-      { pubkey: SystemProgram.programId, isSigner: false, isWritable: false },
-      { pubkey: StakeProgram.programId, isSigner: false, isWritable: false },
     ];
 
+    let data: Buffer;
+    if (ephemeralStake !== undefined && ephemeralStakeSeed !== undefined) {
+      const type = STAKE_POOL_INSTRUCTION_LAYOUTS.IncreaseAdditionalValidatorStake;
+      data = encodeData(type, { lamports, transientStakeSeed, ephemeralStakeSeed });
+      keys.push({ pubkey: ephemeralStake, isSigner: false, isWritable: true });
+    } else {
+      const type = STAKE_POOL_INSTRUCTION_LAYOUTS.IncreaseValidatorStake;
+      data = encodeData(type, { lamports, transientStakeSeed });
+    }
+
+    keys.push(
+      ...[
+        { pubkey: transientStake, isSigner: false, isWritable: true },
+        { pubkey: validatorStake, isSigner: false, isWritable: false },
+        { pubkey: validatorVote, isSigner: false, isWritable: false },
+        { pubkey: SYSVAR_CLOCK_PUBKEY, isSigner: false, isWritable: false },
+        { pubkey: SYSVAR_RENT_PUBKEY, isSigner: false, isWritable: false },
```

Comment:
The "additional" form doesn't include the rent sysvar, so you'll have to do the same trick as in decrease.  In order to avoid issues like that, rather than combining the two instruction creators into one function call, how about keeping them separate?